### PR TITLE
TST: Clean up pickle compression tests

### PR DESCRIPTION
**Pull request by reidy-p**
xref #19226

Move compression fixture to top-level ``conftest.py``

Clean-up compression tests in ``pandas/tests/io/test_pickle.py`` and add zip to the ``decompress_file`` testing utility

Make some minor adjustments to the json and csv compression tests to skip zip compression when it's not valid

### CODE REVIEW

File: pandas/tests/frame/test_to_csv.py [hunk 1]
```diff
@@ -925,22 +925,23 @@ def test_to_csv_compression(self, compression):
                         [12.32112, 123123.2, 321321.2]],
                        index=['A', 'B'], columns=['X', 'Y', 'Z'])
 
-        with ensure_clean() as filename:
+        if compression != "zip":
```
Comment (jreback) [inline pandas/tests/frame/test_to_csv.py]: you could do

```
if compression == 'zip':
    pytest.skip("not zip compression test for to_csv")

....
```

File: pandas/tests/io/json/test_compression.py [hunk 2]
```diff
@@ -10,14 +10,15 @@ def test_compression_roundtrip(compression):
                        [12.32112, 123123.2, 321321.2]],
                       index=['A', 'B'], columns=['X', 'Y', 'Z'])
 
-    with tm.ensure_clean() as path:
-        df.to_json(path, compression=compression)
-        assert_frame_equal(df, pd.read_json(path, compression=compression))
+    if compression != 'zip':
+        with tm.ensure_clean() as path:
```
Comment (jreback) [inline pandas/tests/io/json/test_compression.py]: same here (alternatively, you can make *another* fixed w/o zip), maybe ``compression_no_zip`` is better

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. ping on green (or at least all linux builds on travis), seems mac may have some issues.

**Comment by reidy-p** [comment]
@jreback it's green now, thanks!

**Comment by gfyoung** [comment]
Thanks @reidy-p !

**Pull request merged**